### PR TITLE
Feature/112822 112824 add field

### PR DESCRIPTION
**Pull request by nealfennimore**
https://central.tri.be/issues/112822
https://central.tri.be/issues/112824

- Adds in add rule, exception, repeat event component from mocks
- Adds in withBlockCloser HOC to prevent side effects from occurring when clicking on  inner blocks. Removes [DOM parsing](https://github.com/moderntribe/events-gutenberg/compare/task/112809-pro-plugin-hook-2...feature/112822-112824-add-field?expand=1#diff-109063a8ef3f741d29f5824fdb712494L34)
- Consolidates events-pro blocks into a single block

### CODE REVIEW

File: plugins/common/src/modules/hoc/with-block-closer.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+export default ( WrappedComponent ) => {
+	/**
+	 * Prevents clicks on block or blacklisted DOM elements
+	 * from closing the block
+	 *
+	 * @class WithBlockCloser
+	 * @extends {PureComponent}
+	 */
+	class WithBlockCloser extends PureComponent {
```
[the remark this answers is not in the document]
Comment (nealfennimore) [reply plugins/common/src/modules/hoc/with-block-closer.js]: This just ensures that we don't close the block unnecessarily. When we click on the side bar, or on any DOM element within the block, it prevents the block from closing without having to do any of the DOM parent look ups it was doing before. And yes, it was intended to be reused, so we should be able to reuse this wherever in coming plugins

[the remark this answers is not in the document]
Comment (bordoni) [reply plugins/common/src/modules/hoc/with-block-closer.js]: Perfecto. Thanks a ton Neal.

File: plugins/events/src/modules/blocks/event-datetime/container.js [hunk 2]
```diff
@@ -213,4 +175,5 @@ export default compose(
 	withStore(),
 	connect( mapStateToProps, mapDispatchToProps, mergeProps ),
 	withSaveData(),
+	withBlockCloser,
```
Comment (paulmskim) [inline plugins/events/src/modules/blocks/event-datetime/container.js]: @nealfennimore I know this PR is closed, but this should also be applied to the Price block, as we have a dashboard there that opens / closes.

Comment (nealfennimore) [reply plugins/events/src/modules/blocks/event-datetime/container.js]: I'll go ahead and get this in the next PR I do - thanks @paulmskim !

**Pull request merged**